recipes/namedtype/all/conanfile.py

```diff
@@ -17,10 +20,37 @@ def _source_subfolder(self):
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
-        url = self.conan_data["sources"][self.version]["url"]
-        extracted_dir = "NamedType-" + os.path.splitext(os.path.basename(url))[0]
+        extracted_dir = glob.glob("NamedType-*")[0]
         os.rename(extracted_dir, self._source_subfolder)
 
+    @property
+    def _minimum_cpp_standard(self):
+        return 11
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "Visual Studio": "15",
+            "gcc": "5",
+            "clang": "3.9",
+            "apple-clang": "8",
+        }
+
+    def configure(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
+        if not min_version:
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
+                self.name, self.settings.compiler))
+        else:
+            if tools.Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+
```

Comment:
```suggestion
    def configure(self):
        if self.settings.compiler.cppstd:
            tools.check_min_cppstd(self, 11)

    def package_id(self):
        self.info.header_only()

```

Usually, CCI recipes don't check minimum compiler version for C++11.

+ this is a header only lib.

Reply:
Restricting is required because VS14 was not fully supported. Usable compiler for this library is maybe closer to C++14 than to C++11.

If header_only is given, which configuration will be selected for testing the package?

Reply:
usually gcc 4.9 on Linux, but I'm not sure.

Reply:
Checked a few header-only libraries, it was gcc7 and gcc5, so it can be a random choice. I think it is better to provide a C++-17-capable compiler for building header-only libraries. Otherwise, we have no guarantees that the CI test will pass.

Reply:
Well I've tested, and it requires for sure C++14.

Reply:
Great, patched with `14`

Reply:
yes, but forgot to say that with C++14, you have to keep those compilers checks ;)

Reply:
These versions should work:

```python
    @property
    def _compilers_minimum_version(self):
        return {
            "gcc": "5",
            "Visual Studio": "14",
            "clang": "3.4",
            "apple-clang": "5.1",
        }
```